src/DynamoCoreWpf/ViewModels/Watch3D/HelixWatch3DViewModel.cs

```diff
@@ -523,6 +523,28 @@ protected HelixWatch3DViewModel(NodeModel model, Watch3DViewModelStartupParams p
 
             SetupScene();
             InitializeHelix();
+            DynamoSelection.Instance.Selection.CollectionChanged += OnSelectionCollectionChanged;
+        }
+
+        private void OnSelectionCollectionChanged(object sender, NotifyCollectionChangedEventArgs e)
+        {
+            if (Element3DDictionary == null)
+            {
+                sceneItems = new ObservableElement3DCollection();
+                return;
+            }
+
+            var values = Element3DDictionary.Values.ToList();
+            if (Camera != null)
+            {
+                values.Sort(new Element3DComparer(Camera.Position));
+            }
+
+            if (sceneItems == null)
+                sceneItems = new ObservableElement3DCollection();
+
+            sceneItems.Clear();
+            sceneItems.AddRange(values);
```

Comment:
This logic has been lifted off of `UpdateSceneItems`, however, `UpdateSceneItems` seems to be doing some extra work like cycling through the list of elements and updating vertices, triangles, and colors, which I'm not sure if that is necessary in this case. Skipping that saves some time.